crates/ruff/src/rules/pyupgrade/rules/use_pep695_type_alias.rs

```diff
@@ -75,15 +80,78 @@ pub(crate) fn non_pep695_type_alias(checker: &mut Checker, stmt: &StmtAnnAssign)
     //              as type params instead
     let mut diagnostic = Diagnostic::new(NonPEP695TypeAlias { name: name.clone() }, stmt.range());
     if checker.patch(diagnostic.kind.rule()) {
+        let mut visitor = TypeVarNameVisitor {
+            names: vec![],
+            semantic: checker.semantic(),
+        };
+        visitor.visit_expr(value);
+
+        let type_params = if visitor.names.is_empty() {
+            None
+        } else {
+            Some(ruff_python_ast::TypeParams {
+                range: TextRange::default(),
+                type_params: visitor
+                    .names
+                    .iter()
+                    .map(|name| {
+                        TypeParam::TypeVar(TypeParamTypeVar {
+                            range: TextRange::default(),
+                            name: Identifier::new(name.id.clone(), TextRange::default()),
+                            bound: None,
+                        })
+                    })
+                    .collect(),
+            })
+        };
+
         diagnostic.set_fix(Fix::automatic(Edit::range_replacement(
             checker.generator().stmt(&Stmt::from(StmtTypeAlias {
                 range: TextRange::default(),
                 name: target.clone(),
-                type_params: None,
+                type_params,
                 value: value.clone(),
             })),
             stmt.range(),
         )));
     }
     checker.diagnostics.push(diagnostic);
 }
+
+struct TypeVarNameVisitor<'a> {
```

Comment:
Nit: `#[derive(Debug)]` for any structs that can.

Reply:
Cannot be derived

```
error[E0277]: `SemanticModel<'a>` doesn't implement `std::fmt::Debug`
   --> crates/ruff/src/rules/pyupgrade/rules/use_pep695_type_alias.rs:125:5
    |
122 | #[derive(Debug)]
    |          ----- in this derive macro expansion
...
125 |     semantic: &'a SemanticModel<'a>,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `SemanticModel<'a>` cannot be formatted using `{:?}` because it doesn't implement `std::fmt::Debug`
    |
   ```